<!DOCTYPE html>
<html lang="en">

<!-- Head tag -->
<head>

    <meta charset="utf-8"/>
    <meta http-equiv="X-UA-Compatible" content="IE=edge"/>
    <meta name="viewport" content="width=device-width, initial-scale=1"/>

    <!--Description-->

    

    
        <meta name="description" content="Here, I&#34;ll post my lates idea and notes"/>
    

    <!--Author-->
    
        <meta name="author" content="Peiyao Hu"/>
    

    <!--Open Graph Title-->
    
        <meta property="og:title" content="RNA-seq"/>
    

    <!--Open Graph Description-->
    
        <meta property="og:description" content="Here, I&#34;ll post my lates idea and notes"/>
    

    <!--Open Graph Site Name-->
        <meta property="og:site_name" content="Welcome to my new world!"/>

    <!--Type page-->
    
        <meta property="og:type" content="article"/>
    

    <!--Page Cover-->
    
    
        <meta property="og:image" content="https://peiyaohu.github.io.gitimg/home-bg.jpg"/>
    

        <meta name="twitter:card" content="summary_large_image"/>

    

    
        <meta name="twitter:image" content="https://peiyaohu.github.io.gitimg/home-bg.jpg"/>
    

    <!-- Title -->
    
    <title>RNA-seq - Welcome to my new world!</title>

    <!-- Bootstrap Core CSS -->
    <link href="//maxcdn.bootstrapcdn.com/bootstrap/3.3.6/css/bootstrap.min.css" rel="stylesheet"/>

    <!-- Custom CSS -->
    
<link rel="stylesheet" href="/css/style.css">


    <!-- Custom Fonts -->
    <link href="//maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" type="text/css"/>
    <link href="//fonts.googleapis.com/css?family=Lora:400,700,400italic,700italic" rel="stylesheet" type="text/css"/>
    <link href="//fonts.googleapis.com/css?family=Open+Sans:300italic,400italic,600italic,700italic,800italic,400,300,600,700,800" rel="stylesheet" type="text/css"/>

    <!-- HTML5 Shim and Respond.js IE8 support of HTML5 elements and media queries -->
    <!-- WARNING: Respond.js doesn't work if you view the page via file:// -->
    <!--[if lt IE 9]>
    <script src="//oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
    <script src="//oss.maxcdn.com/libs/respond.js/1.4.2/respond.min.js"></script>
    <![endif]-->

    <!-- Gallery -->
    <link href="//cdnjs.cloudflare.com/ajax/libs/featherlight/1.3.5/featherlight.min.css" type="text/css" rel="stylesheet"/>

    <!-- Google Analytics -->
    


    <!-- favicon -->
    

<meta name="generator" content="Hexo 6.3.0"></head>


<body>

    <!-- Menu -->
    <!-- Navigation -->
<nav class="navbar navbar-default navbar-custom navbar-fixed-top">
    <div class="container-fluid">
        <!-- Brand and toggle get grouped for better mobile display -->
        <div class="navbar-header page-scroll">
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>
            <a class="navbar-brand" href="/">Configurable Title</a>
        </div>

        <!-- Collect the nav links, forms, and other content for toggling -->
        <div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
            <ul class="nav navbar-nav navbar-right">
                
                    <li>
                        <a href="/">
                            
                                Home
                            
                        </a>
                    </li>
                
                    <li>
                        <a href="/archives">
                            
                                Archives
                            
                        </a>
                    </li>
                
                    <li>
                        <a href="/tags">
                            
                                Tags
                            
                        </a>
                    </li>
                
                    <li>
                        <a href="/categories">
                            
                                Categories
                            
                        </a>
                    </li>
                
                    <li>
                        <a target="_blank" rel="noopener" href="https://github.com/klugjo/hexo-theme-clean-blog">
                            
                                <i class="fa fa-github fa-stack-2x"></i>
                            
                        </a>
                    </li>
                
            </ul>
        </div>
        <!-- /.navbar-collapse -->
    </div>
    <!-- /.container -->
</nav>

    <!-- Main Content -->
    <!-- Page Header -->
<!-- Set your background image for this header in your post front-matter: cover -->

<header class="intro-header" style="background-image: url('/img/home-bg.jpg')">
    <div class="container">
        <div class="row">
            <div class="col-lg-8 col-lg-offset-2 col-md-10 col-md-offset-1">
                <div class="post-heading">
                    <h1>RNA-seq</h1>
                    
                    <span class="meta">
                        <!-- Date and Author -->
                        
                        
                            2023-05-29
                        
                    </span>
                </div>
            </div>
        </div>
    </div>
</header>

<!-- Post Content -->
<article>
    <div class="container">
        <div class="row">

            <!-- Tags and categories -->
           
                <div class="col-lg-4 col-lg-offset-2 col-md-5 col-md-offset-1 post-tags">
                    
                        


<a href="/tags/NGS-RNA-seq-pipeline/">#NGS; RNA-seq; pipeline</a>


                    
                </div>
                <div class="col-lg-4 col-md-5 post-categories">
                    
                </div>
            

            <!-- Gallery -->
            

            <!-- Post Main Content -->
            <div class="col-lg-8 col-lg-offset-2 col-md-10 col-md-offset-1">
                <h2 id="0-Before-Practical"><a href="#0-Before-Practical" class="headerlink" title="0. Before Practical"></a>0. Before Practical</h2><!--  -->
<h2 id="1-Prepare"><a href="#1-Prepare" class="headerlink" title="1. Prepare"></a>1. Prepare</h2><h3 id="a-Create-dir-and-download-rawseq-amp-amp-public-source"><a href="#a-Create-dir-and-download-rawseq-amp-amp-public-source" class="headerlink" title="a. Create dir and download rawseq &amp;&amp; public source"></a>a. Create dir and download rawseq &amp;&amp; public source</h3><figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#!/bin/bash</span></span><br><span class="line"><span class="function"><span class="title">mcd</span></span>() &#123; <span class="built_in">mkdir</span> <span class="string">&quot;<span class="variable">$@</span>&quot;</span> 2&gt; &gt;(sed s/mkdir/mcd/ 1&gt;&amp;2) &amp;&amp; <span class="built_in">cd</span> <span class="string">&quot;<span class="variable">$_</span>&quot;</span>; &#125;</span><br><span class="line"><span class="comment">### Create working dir:</span></span><br><span class="line">work_dir=/public/home/pyhu/project/xjw/immunity_mut_RNA_seq</span><br><span class="line"><span class="built_in">cd</span> <span class="variable">$work_dir</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># create a dir to save processing files</span></span><br><span class="line">mcd pyhu_process</span><br><span class="line"></span><br><span class="line"><span class="comment"># make subdir to save results/scripts/public data source</span></span><br><span class="line"><span class="built_in">mkdir</span> 0_Results 2_script 3_source</span><br><span class="line"></span><br><span class="line"><span class="comment">### PE data are saved at:</span></span><br><span class="line">seq_dir=/public/home/pyhu/project/xjw/immunity_mut_RNA_seq/rawSeq</span><br></pre></td></tr></table></figure>



<h3 id="b-Download-public-source"><a href="#b-Download-public-source" class="headerlink" title="b. Download public source"></a>b. Download public source</h3><figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># create a sub_dir to save public data source and extract downloads (some extracting-scripts are neglected).</span></span><br><span class="line"><span class="built_in">cd</span> 3_source; mcd OsativaKitaake</span><br><span class="line"><span class="comment"># Downloads info: Transcriptome and hard-masked genome of Kitaake_499 from JGI.</span></span><br><span class="line">curl --cookie jgi_session=/api/sessions/d3c709055e283ed319a42e6fc3c5fedb --output download.20230529.093434.zip -d <span class="string">&quot;&#123;\&quot;ids\&quot;:&#123;\&quot;Phytozome-499\&quot;:[\&quot;5a3972e57ded5e35e94f8a26\&quot;,\&quot;5a3972e87ded5e35e94f8a30\&quot;,\&quot;5a3972e97ded5e35e94f8a32\&quot;]&#125;&#125;&quot;</span> -H <span class="string">&quot;Content-Type: application/json&quot;</span> https://files.jgi.doe.gov/filedownload/</span><br><span class="line"></span><br><span class="line"><span class="comment"># Transcriptome and hard-masked genoem of Nip(release 56) are also downloaded and saved in the sub_dir (OsNip_release56)</span></span><br><span class="line"><span class="built_in">cd</span> ..; mcd OsNip_release56</span><br><span class="line"><span class="comment"># transcriptome/annotation/hard-masked genome</span></span><br><span class="line">wget https://ftp.ensemblgenomes.ebi.ac.uk/pub/plants/release-56/fasta/oryza_sativa/cdna/Oryza_sativa.IRGSP-1.0.cdna.all.fa.gz</span><br><span class="line">wget https://ftp.ensemblgenomes.ebi.ac.uk/pub/plants/release-56/gtf/oryza_sativa/Oryza_sativa.IRGSP-1.0.56.gtf.gz</span><br><span class="line">wget https://ftp.ensemblgenomes.ebi.ac.uk/pub/plants/release-56/fasta/oryza_sativa/dna/Oryza_sativa.IRGSP-1.0.dna_rm.toplevel.fa.gz</span><br></pre></td></tr></table></figure>



<h2 id="2-RNA-seq-quantification-using-Salmon-v-1-4-0"><a href="#2-RNA-seq-quantification-using-Salmon-v-1-4-0" class="headerlink" title="2. RNA-seq quantification using Salmon (v.1.4.0)"></a>2. RNA-seq quantification using Salmon (v.1.4.0)</h2><h3 id="a-Create-working-dir"><a href="#a-Create-working-dir" class="headerlink" title="a. Create working dir"></a>a. Create working dir</h3><figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">source_dir=/public/home/pyhu/project/xjw/immunity_mut_RNA_seq/pyhu_process/3_source/OsativaKitaake</span><br><span class="line">cdna_ref=<span class="variable">$source_dir</span>/annotation/OsativaKitaake_499_v3.1.transcript.fa.gz</span><br><span class="line">gdna_ref=<span class="variable">$source_dir</span>/assembly/OsativaKitaake_499_v3.0.hardmasked.fa.gz</span><br><span class="line">module load salmon/1.4.0</span><br><span class="line">out_dir=/public/home/pyhu/project/xjw/immunity_mut_RNA_seq/pyhu_process/0_Results/4_salmon_kitaake</span><br></pre></td></tr></table></figure>



<h3 id="b-Salmon-1-Prepare-transcriptome-indices"><a href="#b-Salmon-1-Prepare-transcriptome-indices" class="headerlink" title="b. [Salmon-1] Prepare transcriptome indices"></a>b. [Salmon-1] Prepare transcriptome indices</h3><figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br></pre></td><td class="code"><pre><span class="line"><span class="built_in">echo</span> <span class="string">&quot;[1/2] Prepare transcriptome indices (mapping-based mode)&quot;</span> </span><br><span class="line"><span class="comment"># prepare decoy.txt</span></span><br><span class="line"><span class="built_in">echo</span> <span class="string">&quot;prepare decoy.txt&quot;</span></span><br><span class="line"><span class="built_in">echo</span> <span class="string">&quot;gunzip&quot;</span></span><br><span class="line"></span><br><span class="line">gunzip -c <span class="variable">$gdna_ref</span>|grep <span class="string">&quot;^&gt;&quot;</span> | <span class="built_in">cut</span> -d <span class="string">&quot; &quot;</span> -f 1 &gt; <span class="variable">$out_dir</span>/decoys.txt</span><br><span class="line"><span class="built_in">echo</span> <span class="string">&quot;sed&quot;</span></span><br><span class="line">bsub -K -J pre-2 -n 2 -R span[hosts=1] -o %J.out -e %J.err -q q2680v2 <span class="string">&quot;sed -i.bak -e &#x27;s/&gt;//g&#x27; <span class="variable">$out_dir</span>/decoys.txt&quot;</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># prepare gentrome.fa.gz: the genome targets (decoys) should come after the transcriptome targets in the reference</span></span><br><span class="line"><span class="built_in">echo</span> <span class="string">&quot;prepare gentrome.fa.gz&quot;</span></span><br><span class="line"><span class="built_in">echo</span> <span class="string">&quot;cat&quot;</span></span><br><span class="line">bsub -K -J pre-2 -n 2 -R span[hosts=1] -o %J.out -e %J.err -q q2680v2 <span class="string">&quot;cat <span class="variable">$cdna_ref</span> <span class="variable">$gdna_ref</span> &gt; <span class="variable">$out_dir</span>/gentrome.fa.gz&quot;</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># Index</span></span><br><span class="line"><span class="built_in">echo</span> <span class="string">&quot;Index&quot;</span></span><br><span class="line">bsub -K -J index -n 2 -R span[hosts=1] -o <span class="built_in">log</span>/indexkit.out -e <span class="built_in">log</span>/indexkit.err -q q2680v2 <span class="string">&quot;salmon index -t <span class="variable">$out_dir</span>/gentrome.fa.gz \</span></span><br><span class="line"><span class="string">  -d <span class="variable">$out_dir</span>/decoys.txt -p 12 \</span></span><br><span class="line"><span class="string">  -i <span class="variable">$out_dir</span>/salmon_index \</span></span><br><span class="line"><span class="string">  --gencode&quot;</span></span><br></pre></td></tr></table></figure>



<h3 id="c-Salmon-2-Quantifying-in-mapping-based-mode"><a href="#c-Salmon-2-Quantifying-in-mapping-based-mode" class="headerlink" title="c. [Salmon-2] Quantifying in mapping-based mode"></a>c. [Salmon-2] Quantifying in mapping-based mode</h3><figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br></pre></td><td class="code"><pre><span class="line"><span class="built_in">echo</span> <span class="string">&quot;[2/2] Quantifying in mapping-based mode&quot;</span></span><br><span class="line">seq_dir=/public/home/pyhu/project/xjw/immunity_mut_RNA_seq/rawSeq</span><br><span class="line"></span><br><span class="line"><span class="keyword">for</span> i <span class="keyword">in</span> `<span class="built_in">ls</span> <span class="variable">$seq_dir</span>/*1.fq.gz`;<span class="keyword">do</span></span><br><span class="line">gz_index=$(<span class="built_in">basename</span> <span class="variable">$i</span>| sed <span class="string">&#x27;s/_1.fq.gz//&#x27;</span>)</span><br><span class="line"></span><br><span class="line"><span class="built_in">echo</span> <span class="string">&quot;Quantify sample: <span class="variable">$gz_index</span>&quot;</span></span><br><span class="line">bsub -K -J quant_<span class="variable">$gz_index</span> -n 2 -R span[hosts=1] -o <span class="built_in">log</span>/<span class="variable">$&#123;gz_index&#125;</span>.out -e <span class="built_in">log</span>/<span class="variable">$&#123;gz_index&#125;</span>.err -q q2680v2 <span class="string">&quot;salmon quant -i <span class="variable">$out_dir</span>/salmon_index \</span></span><br><span class="line"><span class="string">  -l A \</span></span><br><span class="line"><span class="string">  -1 <span class="variable">$seq_dir</span>/<span class="variable">$&#123;gz_index&#125;</span>_1.fq.gz \</span></span><br><span class="line"><span class="string">  -2 <span class="variable">$seq_dir</span>/<span class="variable">$&#123;gz_index&#125;</span>_2.fq.gz \</span></span><br><span class="line"><span class="string">  --validateMappings \</span></span><br><span class="line"><span class="string">  -p 6\</span></span><br><span class="line"><span class="string">  --seqBias \</span></span><br><span class="line"><span class="string">  -o <span class="variable">$out_dir</span>/transcripts_quant&quot;</span></span><br><span class="line"><span class="keyword">done</span></span><br></pre></td></tr></table></figure>

                
            </div>

            <!-- Comments -->
            
                <div class="col-lg-8 col-lg-offset-2 col-md-10 col-md-offset-1">
                    


                </div>
            
        </div>
    </div>
</article>

    <!-- Footer -->
    <hr />

<!-- Footer -->
<footer>
    <div class="container">
        <div class="row">
            <div class="col-lg-8 col-lg-offset-2 col-md-10 col-md-offset-1">
                <ul class="list-inline text-center">
                    

                    

                    
                        <li>
                            <a href="https://github.com/klugjo/hexo-theme-clean-blog" target="_blank">
                                <span class="fa-stack fa-lg">
                                    <i class="fa fa-circle fa-stack-2x"></i>
                                    <i class="fa fa-github fa-stack-1x fa-inverse"></i>
                                </span>
                            </a>
                        </li>
                    

                    

                    

                    
                </ul>
                <p class="copyright text-muted">&copy; 2023 Peiyao Hu<br></p>
                <p class="copyright text-muted">Original Theme <a target="_blank" href="http://startbootstrap.com/template-overviews/clean-blog/">Clean Blog</a> from <a href="http://startbootstrap.com/" target="_blank">Start Bootstrap</a></p>
                <p class="copyright text-muted">Adapted for <a target="_blank" href="https://hexo.io/">Hexo</a> by <a href="http://www.codeblocq.com/" target="_blank">Jonathan Klughertz</a></p>
            </div>
        </div>
    </div>
</footer>


    <!-- After footer scripts -->
    
<!-- jQuery -->
<script src="//code.jquery.com/jquery-2.1.4.min.js"></script>

<!-- Bootstrap -->
<script src="//maxcdn.bootstrapcdn.com/bootstrap/3.3.6/js/bootstrap.min.js"></script>

<!-- Gallery -->
<script src="//cdnjs.cloudflare.com/ajax/libs/featherlight/1.3.5/featherlight.min.js" type="text/javascript" charset="utf-8"></script>

<!-- Disqus Comments -->



</body>

</html>